### ISSUE TITLE
Implement Signal Performance Forecasting

### ISSUE THREAD

**Issue by Mathews-25** [AgesEmpire/StellarSwipe-Backends#110]
**Context:**
Users want predictive insights on signal performance based on historical patterns and market conditions.

**Problem:**
Build ML-based forecasting service that predicts signal success probability and expected P&L.

**What Done Looks Like:**
- Signal success probability prediction
- Expected P&L range estimation
- Confidence intervals
- Model training on historical data
- Real-time prediction API

**Folder Structure:**
```
src/
├── ml/
│   ├── forecasting/
│   │   ├── signal-forecasting.service.ts
│   │   ├── model-training.service.ts
│   │   └── feature-engineering.service.ts
│   ├── models/
│   │   └── signal-predictor.model.ts
│   └── dto/
│       ├── forecast-request.dto.ts
│       └── forecast-result.dto.ts
```

**Implementation Guidelines:**
- Train model on historical signal outcomes
- Features: provider reputation, asset volatility, market trend, signal confidence
- Use TensorFlow.js or external ML service
- Output: success probability (0-100%), expected P&L range
- Retrain model weekly with new data
- Cache predictions (1 hour)

**Forecast Response:**
```typescript
{
  signalId: string;
  successProbability: number; // 0-100
  expectedPnL: {
    low: number;
    mid: number;
    high: number;
  };
  confidence: number; // 0-100
  basedOnSamples: number;
}
```

**Model Features:**
- Provider win rate
- Provider consistency
- Asset pair volatility
- Market trend (bull/bear)
- Time of day
- Signal confidence score

**Edge Cases:**
- Insufficient training data
- Market regime changes
- Model drift over time

**Validation:**
- Predictions show correlation with outcomes
- Model accuracy > 60%
- Confidence intervals realistic

---
